modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationServiceImpl.java

```diff
@@ -319,7 +319,7 @@ private Application doInstallStoredApplication( final NodeId nodeId )
             return null;
         }
 
-        return installOrUpdateApplication( byteSource, applicationKey, false, false );
+        return installOrUpdateApplication( byteSource, applicationKey, false, true );
```

Comment:
stored app install happens on remote node "because" repository was changed. No need to update it again.

```suggestion
        return installOrUpdateApplication( byteSource, applicationKey, false, false );
```